scripts/codex_autoreview.py

```diff
@@ -0,0 +1,641 @@
+            token=token,
+            params={"page": page, "per_page": per_page},
+        )
+        if not batch:
+            break
+        for it in batch:
+            files.append(
+                ChangedFile(
+                    filename=it["filename"],
+                    sha=it["sha"],
+                    status=it.get("status", "modified"),
+                    patch=it.get("patch"),
+                )
+            )
+        if len(batch) < per_page:
+            break
+        page += 1
+    return files
+
+
+def _get_pr(owner: str, repo: str, pr_number: int, token: str) -> dict[str, Any]:
+    return cast(
+        dict[str, Any],
+        _github_api(
+            "GET",
+            f"https://api.github.com/repos/{owner}/{repo}/pulls/{pr_number}",
+            token=token,
+        ),
+    )
+
+
+def _compose_prompt(guidelines: str, repo_root: Path, changed: list[ChangedFile], pr: dict) -> str:
+    # Include high-level PR context
+    pr_title = pr.get("title") or ""
+    pr.get("body") or ""
+    head_label = pr.get("head", {}).get("label", "")
+    base_label = pr.get("base", {}).get("label", "")
+    head_sha = pr.get("head", {}).get("sha", "")
+    base_sha = pr.get("base", {}).get("sha", "")
+
+    intro = (
+        "You are an autonomous code review assistant.\n"
+        "Carefully read the guidelines and analyze ONLY the provided diffs.\n"
+        "Output exactly the JSON as specified. Do not add fences or extra text.\n"
+    )
+
+    context = (
+        f"PR Title: {pr_title}\n"
+        f"From: {head_label} ({head_sha}) -> To: {base_label} ({base_sha})\n\n"
+        "Important paths:\n"
+        f"- Repo root (absolute): {repo_root}\n"
+        "When returning code_location.absolute_file_path, use the absolute path under this root.\n"
+        "Line ranges must overlap with the provided diff hunks.\n"
+    )
+
+    diffs: list[str] = []
+    for f in changed:
+        if not f.patch:
+            continue
+        diffs.append(
+            f"File: {f.filename}\nStatus: {f.status}\nPatch (unified diff):\n---\n{f.patch}\n"
+        )
+
+    diff_blob = (
+        ("\n" + ("\n" + ("-" * 80) + "\n").join(diffs))
+        if diffs
+        else "\n(no diff patch content available)\n"
+    )
+
+    prompt = (
+        f"{intro}\n\n"
+        "Review guidelines (verbatim):\n"
+        f"{guidelines}\n\n"
+        f"{context}\n"
+        "Changed files and patches:\n"
+        f"{diff_blob}\n\n"
+        "Respond now with the JSON schema output only."
+    )
+    return prompt
+
+
+def _ensure_model_auth() -> None:
+    provider = os.environ.get("CODEX_PROVIDER", "openai").strip().lower()
+    if provider == "openai":
+        if not os.environ.get("OPENAI_API_KEY"):
+            print(
+                "Missing OPENAI_API_KEY for model provider 'openai'. "
+                "Set it as a repository secret and expose to the job.",
+                file=sys.stderr,
+            )
+            sys.exit(2)
+
```

Comment:
[P3] Remove no-op read of PR body in _compose_prompt

In `_compose_prompt`, the expression `pr.get("body") or ""` is evaluated but its result is unused, which is dead code and slightly obscures intent (scripts/codex_autoreview.py, function `_compose_prompt`). Removing it improves readability without changing behavior.